src/main/java/com/revature/services/BatchLogicImpl.java

```diff
@@ -10,57 +17,133 @@
 import org.springframework.transaction.annotation.Isolation;
 import org.springframework.transaction.annotation.Transactional;
 
-import com.revature.domain.Batch;
-import com.revature.domain.Person;
-import com.revature.repositories.BatchRepository;
-
 @Service
 @Transactional(readOnly=false, isolation=Isolation.READ_COMMITTED)
 public class BatchLogicImpl implements BatchLogic {
 
 	@Autowired
 	private BatchRepository dao;
 	
+	@Autowired
+	private PersonLogic personLogic;
+	
+	@Autowired
+	private JsonValidation validation;
+	
+	@PersistenceContext
+    private EntityManager entityManager;
+	
 	@Override
 	public Batch getBatchByName(String batchName) {
 		Batch batch = dao.findByNameIgnoreCase(batchName);
+		
+		if (batch == null) {
+			throw new NotFoundException("Batch with name " + batchName + " not found");
+		}
+		
 		return batch;
 	}
 
 	@Override
 	public Batch getBatchById(int batchId) {
 		Batch batch = dao.findOne(batchId);
-		return batch;
+		
+		if (batch == null) {
+			throw new NotFoundException("Batch with id " + batchId + " not found");
 		}
+		
+		return batch;
+	}
 
 	@Override
-	public void deleteBatch(Batch batchName) {
-		dao.delete(batchName);	
+	public String deleteBatch(Integer id) {
+		Batch batch = getBatchById(id);
+		dao.delete(batch);
+		return "Batch: " + id + " - DELETED";
 	}
 	
+	@Override
+	public Batch removePersonsFromBatch(Integer batchId, Integer[] personIds) {
+		
+		validation.validateIntegerArray(personIds);
+		
+		Batch batch = getBatchById(batchId);
+		
+		Set<Person> persons = batch.getPersons();
+		
+		for (Integer personId : personIds) {
+			validation.validateTraineeInBatch(personId, batchId);
+			persons.remove(personLogic.getPersonById(personId));
+		}
+
+		dao.saveAndFlush(batch);
+		entityManager.refresh(batch);
+		
+		return batch;
+	}
 
 	@Override
 	public Page<Batch> getAllBatches(Pageable pageable) {
 		return dao.findAll(pageable);
 	}
 
 	@Override
-	public void createBatch(Batch batchName) {
-		dao.save(batchName);
+	public Batch createBatch(Batch batch) {
+		
+		validation.validateBatchFields(batch);
+		
+		dao.saveAndFlush(batch);
+		entityManager.refresh(batch);
+		return batch;
+	}
+	
+	@Override
+	public Batch addPersonsToBatch(Integer batchId, Integer[] personIds) {
+		
+		validation.validateIntegerArray(personIds);
+		
+		Batch batch = getBatchById(batchId);
+		
+		Set<Person> persons = batch.getPersons();
 		
+		for (Integer personId : personIds) {
+			validation.validateTraineeExists(personId);
+			persons.add(personLogic.getPersonById(personId));
+		}
+
+		dao.saveAndFlush(batch);
+		entityManager.refresh(batch);
+		
+		return batch;
 	}
 
 	@Override
-	public void updateBatch(Batch batchName) {
-		dao.save(batchName);
+	public Batch updateBatch(Batch batchName, Integer id) {
+		
+		Batch batch = getBatchById(id);
+		
+		if(batchName.getName() != null && !batchName.getName().equals("")){
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Move the "" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1132)